### PR TITLE
Chage code to write file content during file creation

### PR DESCRIPTION
**Pull request by astrilet**
We have small bug in the current swift_depositor.rb code. When new swift object is created we use two API calls - one to create an object, another to write data to it. When archiving is turned on on the container, that cases file archive of 0 size been created when data in the file being updated. Fix is simple, just need to write file data in the create method.

### CODE REVIEW

File: lib/pushmi_pullyu/swift_depositer.rb [hunk 1]
```diff
@@ -23,27 +23,30 @@ def deposit_file(file_name, swift_container)
...
-                       era_container.object(file_base_name)
-                     else
-                       era_container.create_object(file_base_name)
-                     end
+    object_exists = era_container.object_exists?(file_base_name)
 
     # Add swift metadata with in accordance to AIP spec:
     # https://docs.google.com/document/d/154BqhDPAdGW-I9enrqLpBYbhkF9exX9lV3kMaijuwPg/edit#
-    metadata = {
+    swift_metadata = {
```
Comment (murny) [inline lib/pushmi_pullyu/swift_depositer.rb]: Calling this swift_metadata is redundant. We know it's swift metadata as we are in the swift_depositer class and interacting with swift. I named it metadata as this is the same naming scheme used in Openstack Ruby gem: https://github.com/ruby-openstack/ruby-openstack/blob/master/lib/openstack/swift/storage_object.rb#L11

File: lib/pushmi_pullyu/swift_depositer.rb [hunk 2]
```diff
@@ -23,27 +23,30 @@ def deposit_file(file_name, swift_container)
...
 
     # ruby-openstack wants all keys of the metadata to be named like "X-Object-Meta-{{Key}}", so update them
-    metadata.transform_keys! { |key| "X-Object-Meta-#{key}" }
+    swift_metadata.transform_keys! { |key| "X-Object-Meta-#{key}" }
 
-    deposited_file.write(File.open(file_name),
-                         { 'etag' => checksum,
-                           'content-type' => 'application/x-tar' }.merge(metadata))
+    # creatre file metadata
+    file_metadata = { 'etag' => checksum,
```
Comment (murny) [inline lib/pushmi_pullyu/swift_depositer.rb]: I would call this `headers` to remain consistent with Ruby Openstack gem (https://github.com/ruby-openstack/ruby-openstack/blob/master/lib/openstack/swift/storage_object.rb#L49) and also because most of this stuff doesn't have anything to do with the file's metadata.

File: lib/pushmi_pullyu/swift_depositer.rb [hunk 3]
```diff
@@ -23,27 +23,30 @@ def deposit_file(file_name, swift_container)
 
     era_container = swift_connection.container(swift_container)
 
-    deposited_file = if era_container.object_exists?(file_base_name)
-                       era_container.object(file_base_name)
-                     else
-                       era_container.create_object(file_base_name)
-                     end
+    object_exists = era_container.object_exists?(file_base_name)
```
Comment (murny) [inline lib/pushmi_pullyu/swift_depositer.rb]: Assigned a variable but never used more then once. Do you need this? Does it improve code readability? Probably not needed then?

File: lib/pushmi_pullyu/swift_depositer.rb [hunk 4]
```diff
@@ -23,27 +23,30 @@ def deposit_file(file_name, swift_container)
...
 
-    deposited_file.write(File.open(file_name),
-                         { 'etag' => checksum,
-                           'content-type' => 'application/x-tar' }.merge(metadata))
+    # creatre file metadata
+    file_metadata = { 'etag' => checksum,
+                      'content-type' => 'application/x-tar' }.merge(swift_metadata)
+
+    # create new file
+    return era_container.create_object(file_base_name, file_metadata, File.open(file_name)) unless object_exists
```
Comment (murny) [inline lib/pushmi_pullyu/swift_depositer.rb]: Using return makes this hard to follow as you are abrupting the code flow and returning early. Also if someone comes into this file and wonders what this returns and sees the deposited_file variable being returned at the bottom, they will make the wrong assumption that this is what gets returned on a new file upload. This could all be simplified as such:

```ruby

    if era_container.object_exists?(file_base_name)
      deposited_file = era_container.object(file_base_name)
      deposited_file.write(File.open(file_name), file_metadata)
    else
      deposited_file = era_container.create_object(file_base_name, file_metadata, File.open(file_name))
    end

    deposited_file
```
Therefore you are always returning the same thing and the code flows much better (instead of being interrupted early and returning before it hits the end block)

File: lib/pushmi_pullyu/swift_depositer.rb [hunk 5]
```diff
@@ -23,27 +23,30 @@ def deposit_file(file_name, swift_container)
...
     }
 
     # ruby-openstack wants all keys of the metadata to be named like "X-Object-Meta-{{Key}}", so update them
-    metadata.transform_keys! { |key| "X-Object-Meta-#{key}" }
+    swift_metadata.transform_keys! { |key| "X-Object-Meta-#{key}" }
 
-    deposited_file.write(File.open(file_name),
-                         { 'etag' => checksum,
-                           'content-type' => 'application/x-tar' }.merge(metadata))
+    # creatre file metadata
```
Comment (murny) [inline lib/pushmi_pullyu/swift_depositer.rb]: Majority of these little code comments aren't needed. Your code should be self explanatory. Telling me `era_container.create_object` creates a new file or creating a hash for file_metadata and telling me `create file metdata` is not needed as the code tells you this and is self explanatory.

Also you spelt `create` wrong here.

File: lib/pushmi_pullyu/swift_depositer.rb (hunk 1 above)
Comment (astrilet) [inline lib/pushmi_pullyu/swift_depositer.rb]: OK will change that back

File: lib/pushmi_pullyu/swift_depositer.rb (hunk 2 above)
Comment (astrilet) [inline lib/pushmi_pullyu/swift_depositer.rb]: good idea, will fix

File: lib/pushmi_pullyu/swift_depositer.rb (hunk 3 above)
Comment (astrilet) [inline lib/pushmi_pullyu/swift_depositer.rb]: will modify code as suggested

File: lib/pushmi_pullyu/swift_depositer.rb (hunk 4 above)
Comment (astrilet) [inline lib/pushmi_pullyu/swift_depositer.rb]: will fix as suggested
